### PR TITLE
Update and restructure content

### LINKED ISSUE #808: Move Getting Started sub-sections under How To/Tutorial as appropriate
It's not currently clear what the split of information between Getting Started and How To is.  Similarly, the distinction between 'Getting Started with building a dapp' and 'Getting Started with integrating MM into your dapp' isn't made clear currently.  

I'm proposing to move the 'Getting Started with building a dapp' topics to the tutorial section and content related to integrated the MM wallet into the How To section.

### LINKED ISSUE #809: Update the Introduction to focus on what integrating MM enables
The Introduction is currently focused on the technology (eg, What is the Ethereum Provider API) rather than what it enables (access to the blockchain from your dapp).

### PR DESCRIPTION
**Pull request by alexandratran**
- Restructure content to de-emphasize beginner/generic dapp content and highlight SDK as recommended method. Add recommended topics for beginners vs experienced devs.
- Add and edit some related conceptual content. See more in these PRs: #859 #860 
- Rephrase and reorganize some topics to be more task-oriented
- Update various links, file names, configuration, and redirects

This PR follows the [MM dev docs restructure proposal](https://docs.google.com/document/d/1iKJNTn-iHqGLJYegZIY4OmYP8U6LFGkn-mDRjQ2cYSg/edit?usp=sharing).

(Note: it's easiest to do all restructuring and renaming files in one go, so we can simplify the redirects.)

Fixes #809 
Fixes #808 

Preview: https://docs.metamask.io/restructure/wallet/

### CODE REVIEW

File: wallet/how-to/interact-with-smart-contracts.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 description: Enable your dapp to interact with smart contracts.
-sidebar_position: 3
+sidebar_position: 5
 ---
 
 # Interact with smart contracts
```
Comment (rajkaramchedu) [inline wallet/how-to/interact-with-smart-contracts.md:6]: The list items here have periods at the end. If this is intended, ignore this comment.

**Review by rajkaramchedu — APPROVED** [review]
Looks and reads great.

File: wallet/assets/web3-architecture.png
Comment (vandan) [inline wallet/assets/web3-architecture.png:1]: User's keys should probably go in the box for Extension & Mobile.

Comment (alexandratran) [reply wallet/assets/web3-architecture.png:1]: @vandan, @danfinlay approved this version of the diagram (see [this Slack thread](https://consensys.slack.com/archives/C02GENU377A/p1689015069214429?thread_ts=1688778862.377129&cid=C02GENU377A)). My reasoning was that it simplified the diagram, and also if I understand correctly, not all RPC calls require user's keys (e.g., standard eth calls like `eth_getBlock...`). What do you think?

Comment (vandan) [reply wallet/assets/web3-architecture.png:1]: I would interpret lines to the end-user as representing some interaction between them and the MetaMask clients or SDK UI. If that's the case, the only time keys are involved is during user onboarding (they either generate a set of keys using MM or import existing keys). Even when the user's keys are required for an RPC call, they're already in the MetaMask clients. The user is just approving the use of their keys through the MetaMask client.

File: README.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # MetaMask developer documentation
 
 This is the MetaMask developer documentation repository.
-The documentation site is hosted at [docs.metamask.io](https://docs.metamask.io/wallet), and it's
+The documentation site is hosted at [`docs.metamask.io`](https://docs.metamask.io/wallet), and it's
```
Comment (vandan) [inline README.md:4]:
```suggestion
The documentation site is hosted at [`docs.metamask.io`](https://docs.metamask.io/), and it's
```

**Comment by vandan** [comment]
Developers following the "Get started building a dapp" path from the home page are led to a pretty confusing sequence that doesn't match the description. 

By taking them to setup steps, that content doesn't lead them to the tutorial where they actually build a dapp. We should either revert by sending them to the tutorial where there can be references to the pre-requisite setup steps or establish more linkage between the setup steps and the tutorial.

**Comment by alexandratran** [comment]
@vandan

> Developers following the "Get started building a dapp" path from the home page are led to a pretty confusing sequence that doesn't match the description.
> 
> By taking them to setup steps, that content doesn't lead them to the tutorial where they actually build a dapp. We should either revert by sending them to the tutorial where there can be references to the pre-requisite setup steps or establish more linkage between the setup steps and the tutorial.

The first paragraph in the first page of the "Get started building a dapp" section refers explicitly to the tutorial: https://docs.metamask.io/restructure/wallet/how-to/get-started-building/set-up-dev-environment/

Is this not enough linkage?

File: wallet/how-to/connect/set-up-sdk/index.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+---
+sidebar_position: 1
+---
+
```
Comment (BelfordZ) [inline wallet/how-to/connect/set-up-sdk/index.md:4]: Are we able to make this autogenerate the cards / items below with a simlar file as [wallet/how-to/connect/_category_.json](https://github.com/MetaMask/metamask-docs/pull/840/files#diff-b404e0f9ae143b54c920baf72661855ec09c300f25b578ba997de6ba004eec25) ? Not sure, but I'd imagine that the `description` part of the _category_.json would support markdown. The only thing that would be missing would be the note at the bottom I think?

reference: https://docusaurus.io/docs/sidebar/autogenerated#autogenerated-sidebar-metadata

Comment (alexandratran) [reply wallet/how-to/connect/set-up-sdk/index.md:4]: The autogenerated category cards by default don't support displaying bulleted lists with different links, which is the primary need for this page.

I slightly modified the style of the SDK cards to better match the category cards. Let me know if there's a better solution!

**Comment by BelfordZ** [comment]
Overall it looks great, and I like the structure / documentation guideline that [you linked](https://documentation.divio.com/).

The PR is a bit on the big side, making it more difficult to review. If possible, could you split out changes that aren't specifically required as part of this content restructuring? For example, perhaps changes [like these](https://github.com/MetaMask/metamask-docs/pull/840/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R41) could be split out to a seperate PR.

File: wallet/how-to/connect/set-up-sdk/index.md (hunk 3 above)
Comment (alexandratran) [reply wallet/how-to/connect/set-up-sdk/index.md:4]: Swizzling those components to achieve what we currently have would probably take me a long time to figure out, as I'm not experienced with React. I don't think any of the other category pages would benefit from such a feature currently. Could we stick with my workaround for this PR and return to this issue in the future? Or if someone else is willing to experiment with that, or let me know of the specific solution, that's fine too.

**Comment by vandan** [comment]
> @vandan
> 
> > Developers following the "Get started building a dapp" path from the home page are led to a pretty confusing sequence that doesn't match the description.
> > By taking them to setup steps, that content doesn't lead them to the tutorial where they actually build a dapp. We should either revert by sending them to the tutorial where there can be references to the pre-requisite setup steps or establish more linkage between the setup steps and the tutorial.
> 
> The first paragraph in the first page of the "Get started building a dapp" section refers explicitly to the tutorial: https://docs.metamask.io/restructure/wallet/how-to/get-started-building/set-up-dev-environment/
> 
> Is this not enough linkage?
> 
> Also, in our docs, "Get started building a dapp" refers to the beginner/general dapp setup content, which is separate from the more MM-specific content (as described in the [MM dev docs restructure proposal](https://docs.google.com/document/d/1iKJNTn-iHqGLJYegZIY4OmYP8U6LFGkn-mDRjQ2cYSg/edit?usp=sharing)). Can you expand on what is a confusing sequence / what doesn't match what description?

I see the reference to the tutorial now. Yeah, I was just expecting there to be a sequential flow that leads developers from setup to tutorial (for example if they just keep reading through each page and choosing "next"). I think it could be improved by highlighting the tutorial path at the end of setup steps vs mentioning it only at the beginning.

**Comment by vandan** [comment]
@shanejonas I assigned you to this one to share any thoughts you have on the adjusted content structure.
(there's not as much to technically review, but I thought you might be interested in the new onboarding paths)

**Review by vandan — APPROVED** [review]
I've reviewed the content structure and copy changes. Overall these changes provide an improved set of paths for developers with distinct needs. The additional concepts sections are generally helpful to all developers as well.

**Pull request merged**